test/test_tensorboard.py

```diff
@@ -0,0 +1,454 @@
+                                                     "arctanx": np.arctan(n_iter)}, n_iter)
+            x = np.zeros((32, 3, 64, 64))  # output from network
+            writer.add_images('Image', x, n_iter)  # Tensor
+            writer.add_image_with_boxes('imagebox',
+                                        np.zeros((3, 64, 64)),
+                                        np.array([[10, 10, 40, 40], [40, 40, 60, 60]]),
+                                        n_iter)
+            x = np.zeros(sample_rate * 2)
+
+            writer.add_audio('myAudio', x, n_iter)
+            writer.add_video('myVideo', np.random.rand(16, 48, 1, 28, 28).astype(np.float32), n_iter)
+            # writer.add_text('Text', 'text logged at step:' + str(n_iter), n_iter)
+            # writer.add_text('markdown Text', '''a|b\n-|-\nc|d''', n_iter)
+            writer.add_histogram('hist', np.random.rand(100, 100), n_iter)
+            # writer.add_pr_curve('xoxo', np.random.randint(2, size=100), np.random.rand(
+            #     100), n_iter)  # needs tensorboard 0.4RC or later
+            # writer.add_pr_curve_raw('prcurve with raw data', true_positive_counts,
+            #                         false_positive_counts,
+            #                         true_negative_counts,
+            #                         false_negative_counts,
+            #                         precision,
+            #                         recall, n_iter)
+            # export scalar data to JSON for external processing
+            writer.export_scalars_to_json("./all_scalars.json")
+
+
+def test_linting():
+    import subprocess
+    subprocess.check_output(['flake8', 'tensorboardX'])
+
+
+class SummaryWriterTest(unittest.TestCase):
+    def test_summary_writer_ctx(self):
+        # after using a SummaryWriter as a ctx it should be closed
+        with SummaryWriter(filename_suffix='.test') as writer:
+            writer.add_scalar('test', 1)
+        assert writer.file_writer is None
+
+    def test_summary_writer_close(self):
+        # Opening and closing SummaryWriter a lot should not run into
+        # OSError: [Errno 24] Too many open files
+        passed = True
+        try:
+            writer = SummaryWriter()
+            writer.close()
+        except OSError:
+            passed = False
+
+        assert passed
+
+    def test_windowsPath(self):
+        dummyPath = "C:\\Downloads\\fjoweifj02utj43tj430"
+        with SummaryWriter(dummyPath) as writer:
+            writer.add_scalar('test', 1)
+        import shutil
+        shutil.rmtree(dummyPath)
+
+    def test_pathlib(self):
+        import sys
+        if sys.version_info.major == 2:
+            import pathlib2 as pathlib
+        else:
+            import pathlib
+        p = pathlib.Path('./pathlibtest')
+        with SummaryWriter(p) as writer:
+            writer.add_scalar('test', 1)
+        import shutil
+        shutil.rmtree(str(p))
+
+
+class EmbeddingTest(unittest.TestCase):
```

Comment:
Please change the class to be `class TestTensorBoardEmbedding(unittest.TestCase):`